code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,149 @@
+/**
+  *
+  * Allows the parent to act similarly to the Altar of Gods with modularity. Invoke and Sect Selection is done via attacking with a bible. This means you cannot sacrifice Bibles (you shouldn't want to do this anyways although now that I mentioned it you probably will want to).
+  *
+  */
+/datum/component/religious_tool
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Enables access to the global sect directly
+	var/datum/religion_sect/easy_access_sect
+	/// What extent do we want this religious tool to act? In case you don't want full access to the list. Generated on New
+	var/operation_flags
+	/// The rite currently being invoked
+	var/datum/religion_rites/performing_rite
+	///Sets the type for catalyst
+	var/catalyst_type = /obj/item/storage/book/bible
+	///Enables overide of COMPONENT_NO_AFTERATTACK, not recommended as it means you can potentially cause damage to the item using the catalyst.
+	var/force_catalyst_afterattack = FALSE
+
+/datum/component/religious_tool/Initialize(_flags = ALL, _force_catalyst_afterattack = FALSE, override_catalyst_type)
+	. = ..()
+	SetGlobalToLocal() //attempt to connect on start in case one already exists!
+	operation_flags = _flags
+	force_catalyst_afterattack = _force_catalyst_afterattack
+	if(override_catalyst_type)
+		catalyst_type = override_catalyst_type
+
+/datum/component/religious_tool/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_RELIGION_NEWSECT, .proc/SetGlobalToLocal)
```

Comment:
you're listening to a signal you send inside this component, kind of redundant no?

Reply:
If i'm understanding your Q correctly, not if the tool is made pre sect-select (see altar of gods).

Reply:
what i'm saying is; you have access to this proc once the sect is made; registering for the signal is not needed as you can just call SetGlobalToLocal once the sect is made.